debug_module.tex

```diff
@@ -54,55 +54,55 @@ \section{Debug Module Interface (DMI)} \label{dmi}
 it is possible to debug programs from the first instruction executed.
 The Debug Module's own state and registers should only be
 reset at power-up and while
-\Fdmactive in \Rdmcontrol is 0.
+\FdmDmcontrolDmactive in \RdmDmcontrol is 0.
 The halt state of harts should be
-maintained across system reset provided that \Fdmactive is 1,
+maintained across system reset provided that \FdmDmcontrolDmactive is 1,
 although trigger CSRs may be cleared.
 
 Due to clock and power domain crossing issues,
 it may not be possible to perform arbitrary DMI accesses across
 system reset.
-While \Fndmreset or any external reset is asserted, the only supported DM
-operations are reading and writing \Rdmcontrol. The behavior of other accesses
+While \FdmDmcontrolNdmreset or any external reset is asserted, the only supported DM
+operations are reading and writing \RdmDmcontrol. The behavior of other accesses
 is undefined.
 
-There is no requirement on the duration of the assertion of \Fndmreset.
-The implementation must ensure that a write of \Fndmreset to 1 followed by
-a write of \Fndmreset to 0 triggers system reset. The system may take
-an arbitrarily long time to come out of reset, as reported by \Fallunavail,
-\Fanyunavail.
+There is no requirement on the duration of the assertion of \FdmDmcontrolNdmreset.
+The implementation must ensure that a write of \FdmDmcontrolNdmreset to 1 followed by
+a write of \FdmDmcontrolNdmreset to 0 triggers system reset. The system may take
+an arbitrarily long time to come out of reset, as reported by \FdmDmstatusAllunavail,
+\FdmDmstatusAnyunavail.
 
 Individual harts (or several at once) can be reset by selecting them, setting
-and then clearing \Fhartreset. In this case an implementation may reset more
+and then clearing \FdmDmcontrolHartreset. In this case an implementation may reset more
 harts than just the ones that are selected. The debugger can discover which
-other harts are reset (if any) by selecting them and checking \Fanyhavereset
-and \Fallhavereset.
+other harts are reset (if any) by selecting them and checking \FdmDmstatusAnyhavereset
+and \FdmDmstatusAllhavereset.
 
 When harts have been reset, they must set a sticky {\tt havereset} state bit.
 The conceptual {\tt havereset} state bits can be read for selected harts in
-\Fanyhavereset and \Fallhavereset in \Rdmstatus.
+\FdmDmstatusAnyhavereset and \FdmDmstatusAllhavereset in \RdmDmstatus.
 These bits must be set regardless of the cause of the reset.
 The {\tt havereset} bits for the selected harts
-can be cleared by writing 1 to \Fackhavereset in \Rdmcontrol.
+can be cleared by writing 1 to \FdmDmcontrolAckhavereset in \RdmDmcontrol.
 The {\tt havereset} bits may or may not be cleared
-when \Fdmactive is low.
+when \FdmDmcontrolDmactive is low.
 
-When a hart comes out of reset and \Fhaltreq or \Fresethaltreq are set, the
+When a hart comes out of reset and \FdmDmcontrolHaltreq or \Fresethaltreq are set, the
```

Comment:
How did you decide which registers and fields were changed to the fully qualified name and which weren't?  `\Fhaltreq` was changed but `\Fresethaltreq` was not.  And `\Fhartsel` vs. `\FdmDmcontrolHasel`

Reply:
I changed every macro that was autogenerated.

resethaltreq isn't a bit in a register. These bits exists somewhere in the DM, and are accessed by dmcontrol.setresethaltreq and dmcontrol.clrresethaltreq.

hartsel is similar. It's the combination of dmcontrol.hartsello and dmcontrol.hartselhi.